### ISSUE TITLE
Attention on empty tensors fails from xformers>=0.0.17

### ISSUE THREAD

**Issue by samiede** [facebookresearch/xformers#852]
# 🐛 Bug

The following line in `attention_processor.py` (L1034) throws a `RuntimeError: CUDA error: invalid argument` error in xformer version > 0.0.16 when called with empty `query`, `key` and `value` tensors.

```
        hidden_states = xformers.ops.memory_efficient_attention(
            query, key, value, attn_bias=attention_mask, op=self.attention_op, scale=attn.scale
        )
```

## To Reproduce

```
from diffusers import AutoencoderKL

vae = AutoencoderKL.from_pretrained("runwayml/stable-diffusion-v1-5").to('cuda')

result = vae.encode([])

```

```
/opt/sd/lib/python3.10/site-packages/diffusers/models/unet_2d_blocks.py:538: in forward
    hidden_states = attn(hidden_states, temb=temb)
/opt/sd/lib/python3.10/site-packages/torch/nn/modules/module.py:1501: in _call_impl
    return forward_call(*args, **kwargs)
/opt/sd/lib/python3.10/site-packages/diffusers/models/attention_processor.py:322: in forward
    return self.processor(
/opt/sd/lib/python3.10/site-packages/diffusers/models/attention_processor.py:1034: in __call__
    hidden_states = xformers.ops.memory_efficient_attention(
/opt/sd/lib/python3.10/site-packages/xformers/ops/fmha/__init__.py:193: in memory_efficient_attention
    return _memory_efficient_attention(
/opt/sd/lib/python3.10/site-packages/xformers/ops/fmha/__init__.py:291: in _memory_efficient_attention
    return _memory_efficient_attention_forward(
/opt/sd/lib/python3.10/site-packages/xformers/ops/fmha/__init__.py:311: in _memory_efficient_attention_forward
    out, *_ = op.apply(inp, needs_gradient=False)
/opt/sd/lib/python3.10/site-packages/xformers/ops/fmha/cutlass.py:186: in apply
    out, lse, rng_seed, rng_offset = cls.OPERATOR(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <OpOverloadPacket(op='xformers.efficient_attention_forward_cutlass')>
args = ()
kwargs = {'attn_bias': None, 'compute_logsumexp': False, 'custom_mask_type': 0, 'dropout_p': 0.0, ...}

    def __call__(self, *args, **kwargs):
        # overloading __call__ to ensure torch.ops.foo.bar()
        # is still callable from JIT
        # We save the function ptr as the `op` attribute on
        # OpOverloadPacket to access it here.
>       return self._op(*args, **kwargs or {})
E       RuntimeError: CUDA error: invalid configuration argument
```

## Expected behavior

The vae returning an empty tensor (as in xformers 0.0.16).

## Environment

PyTorch Version (e.g., 1.0): 2.0.1+cu118
OS (e.g., Linux): Docker nvidia/cuda:11.8.0-devel-ubuntu22.04
How you installed PyTorch (conda, pip, source): pip
Python version: 3.10.12
CUDA/cuDNN version: V11.8.89
GPU models and configuration: a100

**Comment by danthe3rd** [comment]
Hi @samiede 
This is indeed a bug, I'll try to push a fix next week

**Comment by samiede** [comment]
Perfect! It's a blocker for us to upgrade xformers! :)

**Comment by samiede** [comment]
Hey @danthe3rd , I just upgraded to xformers 0.0.22 but the issue remains, (even if the error is now looking slightly different), is there any updates to this?

Updated error:

```
/opt/sd/lib/python3.10/site-packages/torch/nn/modules/module.py:1501: in _call_impl
    return forward_call(*args, **kwargs)
/opt/sd/lib/python3.10/site-packages/diffusers/models/attention_processor.py:322: in forward
    return self.processor(
/opt/sd/lib/python3.10/site-packages/diffusers/models/attention_processor.py:1034: in __call__
    hidden_states = xformers.ops.memory_efficient_attention(
/opt/sd/lib/python3.10/site-packages/xformers/ops/fmha/__init__.py:223: in memory_efficient_attention
    return _memory_efficient_attention(
/opt/sd/lib/python3.10/site-packages/xformers/ops/fmha/__init__.py:321: in _memory_efficient_attention
    return _memory_efficient_attention_forward(
/opt/sd/lib/python3.10/site-packages/xformers/ops/fmha/__init__.py:341: in _memory_efficient_attention_forward
    out, *_ = op.apply(inp, needs_gradient=False)
/opt/sd/lib/python3.10/site-packages/xformers/ops/fmha/cutlass.py:194: in apply
    return cls.apply_bmhk(inp, needs_gradient=needs_gradient)
/opt/sd/lib/python3.10/site-packages/xformers/ops/fmha/cutlass.py:243: in apply_bmhk
    out, lse, rng_seed, rng_offset = cls.OPERATOR(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <OpOverloadPacket(op='xformers.efficient_attention_forward_cutlass')>
args = ()
kwargs = {'attn_bias': None, 'compute_logsumexp': False, 'custom_mask_type': 0, 'dropout_p': 0.0, ...}

    def __call__(self, *args, **kwargs):
        # overloading __call__ to ensure torch.ops.foo.bar()
        # is still callable from JIT
        # We save the function ptr as the `op` attribute on
        # OpOverloadPacket to access it here.
>       return self._op(*args, **kwargs or {})
E       RuntimeError: CUDA error: invalid configuration argument
E       CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
E       For debugging consider passing CUDA_LAUNCH_BLOCKING=1.
E       Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.
```

**Comment by danthe3rd** [comment]
Hi,
Thanks for the reminder - we're working on a fix. Hopefully we can get it out by end of week :)

**Comment by danthe3rd** [comment]
The fix is ready and we should be able to push it hopefully today

**Comment by samiede** [comment]
Amazing @danthe3rd ! Thank you!

**Comment by danthe3rd** [comment]
This should be fixed in d18d0ef2633989bc24b0db3b0de9f296a2393f10
